editor/plugins/visual_shader_editor_plugin.cpp

```diff
@@ -1808,6 +1808,7 @@ VisualShaderNode *VisualShaderEditor::_add_node(int p_idx, int p_op_idx) {
 			}
 		}
 	}
+	_member_cancel();
```

Comment:
After some debugging: `_member_cancelled` called when the dialog is closed as it should, but always regardless of OK or Cancel is pressed, maybe it's a bug inside AcceptDialog, maybe not. 

Nvm, I think this line should be removed - instead you should try to add **members_dialog->connect("confirmed", callable_mp(this, &VisualShaderEditor::_member_cancel));** to  line 3054

Reply:
If that's the case maybe it would be better to just wait for that bug on the dialog to be fixed instead of this workarround.

Reply:
Regardless of the bug, I think the logic behind my suggestion is correct and _member_cancel should be called at both confirmed and canceled dialog events.

Reply:
I've tested it with that change and the problem is still there. I guess I can push the commit anyway as it seems like a proper change and let proper bug fixing solve the problem with the dialog in the future.

Reply:
@akien-mga is this a known bug? should I create an issue about the misbehaviour of AcceptDialog?

Reply:
> I've tested it with that change and the problem is still there.

That's unexpected but since I cannot test it on Windows where this bug is unmanifested I guess we could trust you and merge it as it is.

Reply:
I think there are a few problems with the window system in AcceptDialog as it prints an error when I start to drag:
```
ERROR: Condition "!get_viewport()->gui_is_dragging()" is true.
   at: set_drag_preview (scene/gui/control.cpp:713)
```
Maybe it isn't properly emitting some signals because of some bugs with the window system.